In [3]:
# conventional way to import pandas
import pandas as pd
# conventional way to import numpy
import numpy as np

from sklearn import metrics
import matplotlib.pyplot as plt

data = pd.read_csv("https://raw.github.com/vincentarelbundock/Rdatasets/master/csv/ISLR/Auto.csv", index_col=0)

print(data.shape)
data.head()

(392, 9)


mpg  cylinders  displacement  horsepower  weight  acceleration  year  \
1  18.0          8         307.0         130    3504          12.0    70   
2  15.0          8         350.0         165    3693          11.5    70   
3  18.0          8         318.0         150    3436          11.0    70   
4  16.0          8         304.0         150    3433          12.0    70   
5  17.0          8         302.0         140    3449          10.5    70   

   origin                       name  
1       1  chevrolet chevelle malibu  
2       1          buick skylark 320  
3       1         plymouth satellite  
4       1              amc rebel sst  
5       1                ford torino

ISLR Auto is a data frame with 392 observations on the following 9 variables:

    mpg: miles per gallon
    cylinders: Number of cylinders between 4 and 8
    displacement: Engine displacement (cu. inches)
    horsepower: Engine horsepower
    weight: Vehicle weight (lbs.)
    acceleration: Time to accelerate from 0 to 60 mph (sec.)
    year: Model year (modulo 100)
    origin: Origin of car (1. American, 2. European, 3. Japanese)
    name: Vehicle name

In [ ]:
from sklearn.model_selection import KFold, cross_val_score
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sklearn.pipeline import Pipeline

x = pd.DataFrame(data.horsepower)
y = data.mpg

model = LinearRegression()
model.fit(x, y)
print(model.intercept_)
print(model.coef_)

In [ ]:
from sklearn.model_selection import cross_val_score
scores = cross_val_score(model, x, y, cv=5)
scores                                              
